cmd/kubectl-gadget/utils/trace.go

```diff
@@ -371,7 +371,15 @@ func getTraceListFromID(traceID string) (*gadgetv1alpha1.TraceList, error) {
 // SetTraceOperation sets the operation of an existing trace.
 // If trace does not exist an error is returned.
 func SetTraceOperation(traceID string, operation string) error {
-	traces, err := getTraceListFromID(traceID)
+	// We have to wait for the previous operation to finish before changing the
+	// trace operation.
```

Comment:
I think this message is a bit misleading. The trace controller removes the annotation **before** executing the operation, hence what this PR is doing is waiting for the controller to start processing the operation.